sw/otbn/crypto/p384_ecdsa_sign.s

```diff
@@ -39,26 +65,46 @@ ecdsa_sign:
        dmem[k1] <= second share of k */
   jal      x1, p384_generate_k
 
-  /* Fill gpp registers with pointers to variables required for p384_sign */
-  /* scalar shares */
-  /*la        x17, k0
-  la        x19, k1
-  /* message */
-  /*la        x6, msg
-  /* signature values */
-  /*la        x14, r
-  la        x15, s
-  /* secret key shares */
-  /*la        x4, d0
-  la        x5, d1
-
   /* Generate the signature. */
   jal      x1, p384_sign
 
   ecall
 
+/**
+ * P-384 ECDSA side-loaded signature generation.
+ * Generate a signature using a private key from a
+ * sideloaded seed.
+ *
+ * @param[in]  dmem[msg]: message to be signed in dmem
+ * @param[out]   dmem[r]: r component of signature
+ * @param[out]   dmem[s]: s component of signature
+ */
+ecdsa_sign_sideload:
+  /* Load keymgr seeds from WSRs.
+       w20,w21 <= seed0
+       w10,w11 <= seed1 */
+  bn.wsrr   w20, KEY_S0_L
+  bn.wsrr   w21, KEY_S0_H
+  bn.wsrr   w10, KEY_S1_L
+  bn.wsrr   w11, KEY_S1_H
+
+  /* Generate secret key d in shares.
+       dmem[d0] <= d0
+       dmem[d1] <= d1 */
+  jal       x1, p384_key_from_seed
+
+  /* Tail-call signature-generation routine. */
+  jal       x1, ecdsa_sign
```

Comment:
nit: since this is a tail-call and `ecdsa_sign` will hit `ecall` instead of returning, there's no need to consume a call stack slot here:
```suggestion
  jal       x0, ecdsa_sign
```

Reply:
Missed that. Thanks for pointing out!